x-pack/plugins/security_solution/server/lib/risk_engine/risk_engine_data_client.ts

```diff
@@ -57,10 +130,209 @@ export class RiskEngineDataClient {
+  }
+
+  private async getLastUpdatedBy({ savedObjectsClient }: SavedObjectsClients) {
+    const configuration = await this.getConfiguration({ savedObjectsClient });
+
+    if (configuration) {
+      return configuration.last_updated_by;
+    }
+
+    return '';
+  }
+
+  private async getCurrentStatus({ savedObjectsClient }: SavedObjectsClients) {
+    const configuration = await this.getConfiguration({ savedObjectsClient });
+
+    if (configuration) {
+      return configuration.enable ? RiskEngineStatus.ENABLED : RiskEngineStatus.DISABLED;
+    }
+
+    return RiskEngineStatus.NOT_INSTALLED;
+  }
+
+  private async getLegacyTransforms({ namespace }: { namespace: string }) {
+    const esClient = await this.options.elasticsearchClientPromise;
+
+    const getTransformStatsRequests: Array<Promise<TransformGetTransformStatsResponse>> = [];
+    [RiskScoreEntity.host, RiskScoreEntity.user].forEach((entity) => {
+      getTransformStatsRequests.push(
+        esClient.transform.getTransformStats({
+          transform_id: getRiskScorePivotTransformId(entity, namespace),
+        })
+      );
+      getTransformStatsRequests.push(
+        esClient.transform.getTransformStats({
+          transform_id: getRiskScoreLatestTransformId(entity, namespace),
+        })
+      );
+    });
+
+    const result = await Promise.allSettled(getTransformStatsRequests);
+
+    const fulfuletGetTransformStats = result
+      .filter((r) => r.status === 'fulfilled')
+      .map((r) => (r as PromiseFulfilledResult<TransformGetTransformStatsResponse>).value);
+
+    const transforms = fulfuletGetTransformStats.reduce((acc, val) => {
+      if (val.transforms) {
+        return [...acc, ...val.transforms];
+      }
+      return acc;
+    }, [] as TransformGetTransformStatsTransformStats[]);
+
+    return transforms;
+  }
+
+  private async getLegacyStatus({ namespace }: { namespace: string }) {
+    const transforms = await this.getLegacyTransforms({ namespace });
+
+    if (transforms.length === 0) {
+      return RiskEngineStatus.NOT_INSTALLED;
+    }
+
+    const notStoppedTransformsExisted = transforms.some((t) => t.state !== 'stopped');
+
+    if (notStoppedTransformsExisted) {
+      return RiskEngineStatus.ENABLED;
+    }
+
+    return RiskEngineStatus.DISABLED;
+  }
+
+  private async getConfigurationSavedObject({ savedObjectsClient }: SavedObjectsClients) {
+    const savedObjectsResponse = await savedObjectsClient.find({
+      type: riskEngineConfigurationTypeName,
+    });
+    return savedObjectsResponse.saved_objects?.[0];
+  }
+
+  private async getConfiguration({
+    savedObjectsClient,
+  }: SavedObjectsClients): Promise<Configuration | null> {
+    try {
+      const savedObjectConfiguration = await this.getConfigurationSavedObject({
+        savedObjectsClient,
+      });
+      const configuration = savedObjectConfiguration?.attributes;
+
+      if (configuration) {
+        return configuration as Configuration;
```

Comment:
I think you can type  `getConfigurationSavedObject`'s return and remove the need for this cast.